File: crm-service/internal/core/domain/ticket.go
package domain

import (
	"time"
)

// TicketStatus represents the status of a ticket
type TicketStatus string

// TicketPriority represents the priority of a ticket
type TicketPriority string

// Ticket statuses
const (
	StatusNew        TicketStatus = "new"
	StatusOpen       TicketStatus = "open"
	StatusInProgress TicketStatus = "in_progress"
	StatusResolved   TicketStatus = "resolved"
	StatusClosed     TicketStatus = "closed"
)

// Ticket priorities
const (
	PriorityLow      TicketPriority = "low"
	PriorityMedium   TicketPriority = "medium"
	PriorityHigh     TicketPriority = "high"
	PriorityCritical TicketPriority = "critical"
)

// Ticket represents a customer support ticket in the CRM system
type Ticket struct {
	ID          string         `json:"id"`
	CustomerID  string         `json:"customer_id"`
	AgentID     *string        `json:"agent_id,omitempty"`
	Subject     string         `json:"subject"`
	Description string         `json:"description"`
	Status      TicketStatus   `json:"status"`
	Priority    TicketPriority `json:"priority"`
	Tags        []string       `json:"tags,omitempty"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	ClosedAt    *time.Time     `json:"closed_at,omitempty"`
}

// TicketEvent represents an event in the history of a ticket
type TicketEvent struct {
	ID        string    `json:"id"`
	TicketID  string    `json:"ticket_id"`
	UserID    string    `json:"user_id"`
	EventType string    `json:"event_type"` // created, status_changed, comment, assigned, closed
	Content   string    `json:"content"`
	Timestamp time.Time `json:"timestamp"`
}

File: frontend/src/types/ticket.ts
import type { Customer } from "./customer"

export type TicketStatus = 'new' | 'open' | 'in_progress' | 'resolved' | 'closed'
export type TicketPriority = 'low' | 'medium' | 'high'

export interface Ticket {
  id: string
  subject: string
  description: string
  status: TicketStatus
  priority: TicketPriority
  customer: Customer
  customerId: string
  assignedToId?: string
  assignedTo?: {
    id: string
    firstName: string
    lastName: string
    email: string
  }
  createdAt: string
  updatedAt: string
}

export interface TicketListResponse {
  tickets: Ticket[]
  total: number
  page: number
  pageSize: number
  hasNextPage: boolean
}

export interface TicketStats {
  new: number
  open: number
  in_progress: number
  resolved: number
  closed: number
  total: number
}

export interface CreateTicketData {
  subject: string
  description: string
  customerId: string
  priority: TicketPriority
  status: TicketStatus
  assignedToId?: string
}

export type UpdateTicketData = Partial<CreateTicketData>